core/llm.go

```diff
@@ -480,6 +467,26 @@ func (llm *LLM) Clone() *LLM {
 	return &cp
 }
 
+func (llm *LLM) Endpoint(ctx context.Context) (*LLMEndpoint, error) {
+	if llm.endpoint != nil {
```

Comment:
I saw what you mentioned about `sync.Once` in the description, but you can use [OnceValues](https://pkg.go.dev/sync#OnceValues) to help w/ errors. Or even just a mutex would be fine here

Reply:
lmao i read through [the proposal to add that](https://github.com/golang/go/issues/56102) and several others and came to the conclusion that none of them ever shipped... clearly i misunderstood the final outcome XD 

will fix, oncevalues is exactly what i needed.